src/Services/RegistrationsInfoCacheService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Core.Services;
+using Lykke.Service.Registration;
+using MessagePack;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Services
+{
+    public class RegistrationsInfoCacheService : IRegistrationsInfoCacheService
```

Comment:
will be better to implement it as decorator for ```ILykkeRegistrationClient```

Here  https://github.com/LykkeCity/LykkePublicAPI/blob/master/src/LykkePublicAPI/Modules/ApiModule.cs#L75 you can see how to register decorator